iceoryx_posh/test/moduletests/test_roudi_process.cpp

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iceoryx_posh/iceoryx_posh_types.hpp"
+#include "iceoryx_posh/internal/roudi/roudi_process.hpp"
+#include "iceoryx_posh/mepoo/mepoo_config.hpp"
+#include "iceoryx_posh/roudi/memory/roudi_memory_interface.hpp"
+#include "iceoryx_posh/version/compatibility_check_level.hpp"
+#include "iceoryx_utils/cxx/string.hpp"
+#include "test.hpp"
+
+using namespace ::testing;
+using namespace iox::roudi;
+using namespace iox::popo;
+using namespace iox::runtime;
+using ::testing::Return;
+
+class MqInterfaceUser_Mock : public iox::roudi::RouDiProcess
+{
+  public:
+    MqInterfaceUser_Mock()
+        : iox::roudi::RouDiProcess("TestRoudiProcess", 200, nullptr, true, 0x654321, 255)
+    {
+    }
+    MOCK_METHOD1(sendToMQ, void(MqMessage));
+};
+
+class RouDiProcess_test : public Test
+{
+  public:
+    const iox::ProcessName_t processname = "TestRoudiProcess";
+    int32_t pid = 200;
```

Comment:
I saw that in the declaration of the pid in RouDiProcess the fundamental type `int` is used, this should be changed to `int32_t`

Reply:
Additionally please ensure the correct initialization of all class members in RouDiProcess

Reply:
I modified the code to     
pid_t pid = 200;
iox::mepoo::MemoryManager* payloadMemoryManager;   removed the null pointer

but i am not able to fully understand what is meant by correct initialization of all class members in RouDiProcess

Reply:
it means that the members declared in RouDiProcess class should have a default value at creation:
e.g. `uint64_t m_member{0U};`
@chiranjeevimaddi